File: DSA/Ayush_Mishra_590014844/Day_11_Question_1.py
import string
def is_palindrome(s):
    cleaned = ''.join(c.lower() for c in s if c.isalnum())
    return cleaned == cleaned[::-1]

# Examples
print(is_palindrome("A man, a plan, a canal: Panama"))  # True
print(is_palindrome("race a car"))                      # False

File: DSA/Ayush_Mishra_590014844/Day_11_Question_1.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>
int is_palindrome(const char *str) {
    char cleaned[1000];
    int len = 0;
    for (int i = 0; str[i]; i++) {
        if (isalnum(str[i])) {
            cleaned[len++] = tolower(str[i]);
        }
    }

    for (int i = 0; i < len / 2; i++) {
        if (cleaned[i] != cleaned[len - 1 - i]) return 0;
    }
    return 1;
}

int main() {
    printf("%d\n", is_palindrome("A man, a plan, a canal: Panama")); // 1
    printf("%d\n", is_palindrome("race a car"));                     // 0
}
